test/k2_tree_test.cpp

```diff
@@ -538,6 +538,36 @@ TYPED_TEST(k2_tree_test, serialize_test)
 
 }
 
+#if SDSL_HAS_CEREAL
+template <typename in_archive_t, typename out_archive_t, typename TypeParam>
+void do_serialisation(TypeParam const & l)
+{
+	std::stringstream ss;
+	{
+		out_archive_t oarchive{ss};
+		oarchive(l);
+	}
+
+	{
+		TypeParam in_l{};
+		in_archive_t iarchive{ss};
+		iarchive(in_l);
+		EXPECT_EQ(l, in_l);
+	}
+}
+
+TYPED_TEST(k2_tree_test, cereal)
+{
+	TypeParam k2tree;
+        // ASSERT_TRUE(load_from_file(k2treap, temp_file));
```

Comment:
Should this not be commented in?

Reply:
No, there is no `temp_file` for this test, but I wanted to add a construction.

Reply:
Ok, then let's take a single non-trivial test:

```suggestion
    vector<vector<int>> mat({{1, 0, 0, 0, 1},
        {0, 0, 0, 0, 0},
        {0, 0, 1, 1, 0},
        {0, 0, 0, 0, 0},
        {0, 0, 1, 0, 1}
    });
    k2tree = TypeParam(mat);
```